neurom/fst/tests/test_get_features.py

```diff
@@ -0,0 +1,204 @@
+# Copyright (c) 2015, Ecole Polytechnique Federale de Lausanne, Blue Brain Project
+# All rights reserved.
+#
+# This file is part of NeuroM <https://github.com/BlueBrain/NeuroM>
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+#     1. Redistributions of source code must retain the above copyright
+#        notice, this list of conditions and the following disclaimer.
+#     2. Redistributions in binary form must reproduce the above copyright
+#        notice, this list of conditions and the following disclaimer in the
+#        documentation and/or other materials provided with the distribution.
+#     3. Neither the name of the copyright holder nor the names of
+#        its contributors may be used to endorse or promote products
+#        derived from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY
+# DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+'''Test neurom.fst.get features'''
+
+import os
+import numpy as np
+from nose import tools as nt
+from neurom.core.types import NeuriteType
+from neurom.core.population import Population
+from neurom import fst
+
+
+_PWD = os.path.dirname(os.path.abspath(__file__))
+NRN_FILES = [os.path.join(_PWD, '../../../test_data/h5/v1', f)
+             for f in ('Neuron.h5', 'Neuron_2_branch.h5', 'bio_neuron-001.h5')]
+
+NRNS = fst.load_neurons(NRN_FILES)
+NRN = NRNS[0]
+POP = Population(NRNS)
+
+NEURITES = (NeuriteType.axon,
+            NeuriteType.apical_dendrite,
+            NeuriteType.basal_dendrite,
+            NeuriteType.all)
+
+
+def test_number_of_sections_pop():
+
+    feat = 'number_of_sections'
+
+    nt.assert_items_equal(fst.get(feat, POP), [84, 42, 202])
+
+    nt.assert_items_equal(fst.get(feat, POP, neurite_type=NeuriteType.all),
+                          [84, 42, 202])
+
+    nt.assert_items_equal(fst.get(feat, POP, neurite_type=NeuriteType.axon),
+                          [21, 21, 179])
+
+    nt.assert_items_equal(fst.get(feat, POP, neurite_type=NeuriteType.apical_dendrite),
+                          [21, 0, 0])
```

Comment:
I think if we're returning numbers *per neuron*, it makes sense to return `0` where applicable, because that is the number of things for a given neuron for a given neurite type. I think not to do this would be quite confusing.